File: programmers/[Greedy]/체육복.py
# 체육복

def solution(n, lost, reserve):
    lost_check = [False] * len(lost)
    reserve_check = [False] * len(reserve)
    answer = n - len(lost)

    for i in range(len(lost)):
        for j in range(len(reserve)):
            if lost[i] == reserve[j]:
                lost_check[i] = True
                reserve_check[j] = True
                answer += 1

    for i in range(len(lost)):
        for j in range(len(reserve)):
            if not lost_check[i]:
                if not reserve_check[j]:
                    if abs(lost[i] - reserve[j]) == 1:
                        answer += 1
                        lost_check[i] = True
                        reserve_check[j] = True

    return answer

File: Algorithm-Swift/Algorithm-Swift/Programmers/그리디/체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var answer = n - lost.count
    var lost_stu = Array(repeating: false, count: lost.count)
    var reserve_stu = Array(repeating: false, count: reserve.count)
    
    // 체육복 여벌로 가져온 학생이 도난 당한 경우 체크
    for (ldx,l) in lost.enumerated() {
        for (rdx,r) in reserve.enumerated() {
            if r == l {
                lost_stu[ldx] = true
                reserve_stu[rdx] = true
                answer += 1
            }
        }   
    }
    
    // 체육복 빌려주기 
    for (ldx,l) in lost.enumerated() {
        for (rdx,r) in reserve.enumerated() {
            if abs(r-l) == 1 && !lost_stu[ldx] && !reserve_stu[rdx] {
                lost_stu[ldx] = true
                reserve_stu[rdx] = true
                answer += 1
            }
        }   
    }
    
    return answer
}


// 
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    // MARK: - 체육복 여벌로 가져온 학생이 도난 당한 경우 체크
    var reserved = reserve.filter{ !lost.contains($0) } 
    var losted = lost.filter{ !reserve.contains($0) }

    var lost_stu = Array(repeating: false, count: losted.count)
    var reserve_stu = Array(repeating: false, count: reserved.count)
    var answer = n - losted.count
    
    // 체육복 빌려주기
    for (ldx,l) in losted.enumerated() {
        for (rdx,r) in reserved.enumerated() {
            if abs(r-l) == 1 && !lost_stu[ldx] && !reserve_stu[rdx] {
                lost_stu[ldx] = true
                reserve_stu[rdx] = true
                answer += 1
            }
        }   
    }
    
    return answer
}